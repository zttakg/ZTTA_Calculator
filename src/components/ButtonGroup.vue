
<script>
export default {
  name: "ButtonGroup",

  props: {
    baseClassName: {
      type: String,
      default: "button-group"
    }
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    return (
      <div class={this.baseClassName}>
        {this.$slots.default &&
          // eslint-disable-next-line arrow-parens
          this.$slots.default.map(vnode => {
            const clonedVnode = vnode;
            clonedVnode.data.class = {};
            clonedVnode.data.class["button-group__item"] = true;
            return clonedVnode;
          })}
      </div>
    );
  }
};
</script>
