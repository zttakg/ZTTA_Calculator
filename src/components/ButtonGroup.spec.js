import { shallowMount } from "@vue/test-utils";
import ButtonGroup from "./ButtonGroup";

describe("<ButtonGroup /> component", () => {
  const slots = {
    default: [
      '<button type="button" class="button"></button>',
      '<button type="button" class="button"></button>'
    ]
  };

  it("should render correct without props", () => {
    const wrapper = shallowMount(ButtonGroup, {
      slots
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
