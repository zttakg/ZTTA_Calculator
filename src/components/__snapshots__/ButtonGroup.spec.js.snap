// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> component should render correct without props 1`] = `
<div class="button-group">
  <button type="button" class="button button-group__item"></button>
  <button type="button" class="button button-group__item"></button>
</div>
`;
